### PR TITLE
[JAVA][Feign] Replace Apache oltu with scribejava

### PR DESCRIPTION
**Pull request by hugo-ma-alves**
Hello

This PR is the implementation for the issue https://github.com/OpenAPITools/openapi-generator/issues/8297.

To sum up this PR addresses the following issues:

- Replaces apache OLTU with  scribejava
- Implements Oauth client_credentials grant and password flow. 
I can also implement the implicit flow. But I am not sure how to handle the user redirections in this context. These changes focus more on server to server communication. I Believe that the implicit flow is considered deprecated, it should be replaced by the authorization code grant.

The following git repo contains the Petstore client generated using the code on this PR. The open api spec file is also on the git repo
https://github.com/hugo-ma-alves/openapi-generator-petstore-demo

According with the ticket https://github.com/OpenAPITools/openapi-generator/issues/6158 the support for java 7 is not guaranteed anymore right?  I did some java 8 only code, so it won't work on jre7. But it should be easy to revert to java 7.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

btw, I committed the changes on the samples folder, not sure if they should also be on git. Let me know if I should remove them.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by wing328** [comment]
> According with the ticket #6158 the support for java 7 is not guaranteed anymore right? I did some java 8 only code, so it won't work on jre7. But it should be easy to revert to java 7.

That's correct

**Comment by wing328** [comment]
CircleCI reports the following errors:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.1.1:jar (attach-javadocs) on project petstore-feign: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/feign/src/main/java/org/openapitools/client/ApiClient.java:294: error: lambda expressions are not supported in -source 1.7
[ERROR] .filter(requestInterceptor -> type.isAssignableFrom(requestInterceptor.getClass()))
[ERROR] ^
[ERROR] (use -source 8 or higher to enable lambda expressions)
[ERROR] 
[ERROR] Command line was: /usr/lib/jvm/java-8-openjdk-amd64/jre/../bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/feign/target/apidocs' dir.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :petstore-feign
```

Can you please take a look when you've time?

**Comment by hugo-ma-alves** [comment]
Forgot to remove the support for java 1.7 from the pom 😞 .  Because I introduced lambda expressions it won't work anymore for java<1.8
I pushed some changes, let see if it works.

**Comment by hugo-ma-alves** [comment]
Now it is throwing some timeout error.
```
21:36:20.954 [qtp2074407503-12] INFO  i.s.sample.resource.PetStoreResource - placeOrder ID 1001 STATUS pending
21:37:38.877 [qtp2074407503-10] INFO  i.s.sample.resource.UserResource - createUser ID 11222 STATUS scala-test-username
21:37:39.143 [qtp2074407503-11] INFO  i.s.sample.resource.UserResource - createUsersWithArrayInput
21:37:39.181 [qtp2074407503-14] INFO  i.s.sample.resource.UserResource - createUsersWithListInput
21:37:39.226 [qtp2074407503-15] INFO  i.s.sample.resource.UserResource - createUser ID 4000 STATUS tony
21:37:39.241 [qtp2074407503-14] INFO  i.s.sample.resource.UserResource - updateUser ID 4000 STATUS tony

Too long with no output (exceeded 10m0s): context deadline exceeded```

It looks related with the scala module? Not sure what the problem is in this case.

**Comment by wing328** [comment]
No worry. We can safely remove java6, java7 support in the Feign client

Please update the samples when you've time to fix the following:
```
--- a/samples/client/petstore/java/feign-no-nullable/README.md
+++ b/samples/client/petstore/java/feign-no-nullable/README.md
@@ -75,4 +75,3 @@ It's recommended to create an instance of `ApiClient` per thread in a multithrea
 
 
 
-
diff --git a/samples/client/petstore/java/feign/README.md b/samples/client/petstore/java/feign/README.md
index afd609fe63..fa5dd565da 100644
--- a/samples/client/petstore/java/feign/README.md
+++ b/samples/client/petstore/java/feign/README.md
@@ -75,4 +75,3 @@ It's recommended to create an instance of `ApiClient` per thread in a multithrea
 
 
 
-
Perform git status
On branch feature/replace-apache-oltu-with-scribejava
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/client/petstore/java/feign-no-nullable/README.md
	modified:   samples/client/petstore/java/feign/README.md
```

**Comment by hugo-ma-alves** [comment]
Done, meanwhile I also updated the mustache template for gradle and sbt. They still had references for apache oltu.

**Comment by wing328** [comment]
```
--- a/samples/client/petstore/java/feign-no-nullable/build.gradle
+++ b/samples/client/petstore/java/feign-no-nullable/build.gradle
@@ -116,6 +116,7 @@ dependencies {
     implementation "com.fasterxml.jackson.core:jackson-annotations:$jackson_version"
     implementation "com.fasterxml.jackson.core:jackson-databind:$jackson_databind_version"
     implementation "com.github.joschi.jackson:jackson-datatype-threetenbp:$jackson_threetenbp_version"
+    implementation "com.brsanthu:migbase64:2.2"
     implementation "com.github.scribejava:scribejava-core:$scribejava_version"
     implementation "com.brsanthu:migbase64:2.2"
     implementation 'javax.annotation:javax.annotation-api:1.3.2'
diff --git a/samples/client/petstore/java/feign/build.gradle b/samples/client/petstore/java/feign/build.gradle
index e077363990..637b0f9d64 100644
--- a/samples/client/petstore/java/feign/build.gradle
+++ b/samples/client/petstore/java/feign/build.gradle
@@ -118,6 +118,7 @@ dependencies {
     implementation "com.fasterxml.jackson.core:jackson-databind:$jackson_databind_version"
     implementation "org.openapitools:jackson-databind-nullable:$jackson_databind_nullable_version"
     implementation "com.github.joschi.jackson:jackson-datatype-threetenbp:$jackson_threetenbp_version"
+    implementation "com.brsanthu:migbase64:2.2"
     implementation "com.github.scribejava:scribejava-core:$scribejava_version"
     implementation "com.brsanthu:migbase64:2.2"
     implementation 'javax.annotation:javax.annotation-api:1.3.2'
Perform git status
On branch pull/8318
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/client/petstore/java/feign-no-nullable/build.gradle
	modified:   samples/client/petstore/java/feign/build.gradle

no changes added to commit (use "git add" and/or "git commit -a")

```

As reported by the CircleCI, the samples are not yet updated. Let me know if you need help on that.

**Comment by hugo-ma-alves** [comment]
I merged the master branch into this branch and forgot to update the samples.
All done now.

**Comment by wing328** [comment]
CircleCI tests passed via https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/8173/workflows/620e3e1e-d0ab-466f-ad27-5fbd57bd45a0/jobs/21726

File: modules/openapi-generator-gradle-plugin/gradlew.bat [hunk 1]
```diff
@@ -1,100 +1,100 @@
...
+
+:fail
+rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
+rem the _cmd.exe /c_ return code!
+if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
+exit /b 1
+
+:mainEnd
+if "%OS%"=="Windows_NT" endlocal
+
```
Comment (wing328) [inline modules/openapi-generator-gradle-plugin/gradlew.bat:99]: Looks like this is not the latest `gradlew.bat`. I'll file another PR to update it.

cc @jimschubert

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 2]
```diff
@@ -392,6 +392,10 @@ public void processOpts() {
             forceSerializationLibrary(SERIALIZATION_LIBRARY_JACKSON);
             supportingFiles.add(new SupportingFile("ParamExpander.mustache", invokerFolder, "ParamExpander.java"));
             supportingFiles.add(new SupportingFile("EncodingUtils.mustache", invokerFolder, "EncodingUtils.java"));
+            supportingFiles.add(new SupportingFile("auth/DefaultApi20Impl.mustache", authFolder, "DefaultApi20Impl.java"));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:395]: @hugo-ma-alves quick question. How is the file `"DefaultApi20Impl` named?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:395]: Just saw it: `import com.github.scribejava.core.builder.api.DefaultApi20;`

Please ignore my question.

File: modules/openapi-generator-gradle-plugin/gradlew.bat (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator-gradle-plugin/gradlew.bat:99]: Filed https://github.com/OpenAPITools/openapi-generator/pull/8469

**Comment by pPanda-beta** [comment]
@hugo-ma-alves 
What about the `OAuthFlow`
when people dont want it, should not we hide it?

https://github.com/OpenAPITools/openapi-generator/pull/8318/files#diff-86e0b144aa0784ea6cccf92ec027a7b82407ee0cd823044b82396f7ffe08db92R163

**Comment by hugo-ma-alves** [comment]
@pPanda-beta 
Yep, that is fixed by https://github.com/OpenAPITools/openapi-generator/pull/9472

**Pull request merged**